akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -316,6 +321,7 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
   /**
    * INTERNAL API
    */
+  @volatile
```

Comment:
Why does this need to be volatile now, but not before? It was used before as well? Was that something that was missed? Also isn't this read often (every async event) -> bad for performance? Could we pass it to `onStart` instead or something?

Reply:
Checked once again if `getAsyncCallback` itself must be asynchronous and it should not - as `getAsyncCallback` is called during initialization and executing in one thread for `GraphStage` per my understanding.

Reply:
Yeah, the async callback maintains the "single-threaded illusion"